content/blog/expressions/index.md

```diff
@@ -0,0 +1,193 @@
+
+```rust
+use std::env;
+use std::fs::File;
+use std::path::PathBuf;
+
+/// Create a configuration file for the application at the given path
+fn create_config_file(config_path: Option<PathBuf>) -> Result<File, Box<dyn std::error::Error>> {
+    let path = match config_path {
+        Some(path) => path,
+        None => {
+            if let Some(home_dir) = env::home_dir() {
+                let config_dir = home_dir.join(".config/my_app/");
+                config_dir.to_path_buf()
+            } else {
+                PathBuf::from(".")
+            }
+        }
+    };
+    Ok(setup_config(&path)?)
+}
+
+fn setup_config(path: &PathBuf) -> Result<File, Box<dyn std::error::Error>> {
+    // Implementation details omitted
+    unimplemented!()
+}
+```
+
+Better, right? 
+
+The `else` block in the `None` case is just a fallback if all else fails.
+It gets way too much attention in the original code.
+We can use the `unwrap_or_else` method to make it more concise:
+
+```rust
+use std::env;
+use std::fs::File;
+use std::path::PathBuf;
+
+/// Create a configuration file for the application at the given path
+fn create_config_file(config_path: Option<PathBuf>) -> Result<File, Box<dyn std::error::Error>> {
+    let path = match config_path {
+        Some(path) => path,
+        None => {
+            env::home_dir()
+                .map(|home_dir| home_dir.join(".config/my_app/").to_path_buf())
+                .unwrap_or_else(|| PathBuf::from("."))
+        }
+    };
+    Ok(setup_config(&path)?)
+}
+
+fn setup_config(path: &PathBuf) -> Result<File, Box<dyn std::error::Error>> {
+    // Implementation details omitted
+    unimplemented!()
+}
+```
+
+Now the fallback is just a side note, and the main logic is more prominent.
+
+The `Some(path) => path` match arm looks very simple. Almost too simple? We have already used `unwrap_or_else()` once, let's apply it here as well.
+
+```rust
+use std::env;
+use std::fs::File;
+use std::path::PathBuf;
+
+/// Create a configuration file for the application at the given path
+fn create_config_file(config_path: Option<PathBuf>) -> Result<File, Box<dyn std::error::Error>> {
+    let path = config_path.unwrap_or_else(|| {
+        env::home_dir()
+            .map(|home_dir| home_dir.join(".config/my_app/").to_path_buf())
+            .unwrap_or_else(|| PathBuf::from("."))
+    });
+    Ok(setup_config(&path)?)
+}
+
+fn setup_config(path: &PathBuf) -> Result<File, Box<dyn std::error::Error>> {
+    // Implementation details omitted
+    unimplemented!()
+}
+```
+
+This code could still be improved, but the main point is to show you how expressions can help you refactor your code without a lot of effort. We avoided repetition simply by thinking in terms of expressions.
+
+## Conclusion
+
+Expressions are very powerful. It takes a bit of practice to get used to them, 
+but once you get acquainted with them, they are a joy to work with.
```

Comment:
`to get used to them` and `get acquainted with them` is duplicated

```suggestion
but once you do, they are a joy to work with.
```

Reply:
Good point. I changed that. Kept 'acquainted' because I like the word. 😉